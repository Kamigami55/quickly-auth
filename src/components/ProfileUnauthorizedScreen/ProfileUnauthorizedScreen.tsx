import Link from 'next/link';

import { Button } from '@/components/ui/button';
import { PAGE_URL } from '@/constants/pageUrl';

export function ProfileUnauthorizedScreen() {
  return (
    <div className="flex items-center justify-center p-8">
      <div className="flex flex-col items-center gap-1 text-center">
        <h3 className="text-3xl font-bold tracking-tight">401 Unauthorized</h3>
        <p className="text-muted-foreground">Please login to continue</p>
        <Link href={PAGE_URL.LOGIN}>
          <Button className="mt-4">Login</Button>
        </Link>
      </div>
    </div>
  );
}
