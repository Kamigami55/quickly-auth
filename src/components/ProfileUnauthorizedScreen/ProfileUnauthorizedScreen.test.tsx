import '@testing-library/jest-dom';

import { render } from '@testing-library/react';

import { ProfileUnauthorizedScreen } from './ProfileUnauthorizedScreen';

describe('ProfileUnauthorizedScreen', () => {
  it('renders unauthorized screen unchanged', () => {
    const { container } = render(<ProfileUnauthorizedScreen />);
    expect(container).toMatchSnapshot();
  });
});
