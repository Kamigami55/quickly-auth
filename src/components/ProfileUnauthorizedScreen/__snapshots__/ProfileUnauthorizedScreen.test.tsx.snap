// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileUnauthorizedScreen renders unauthorized screen unchanged 1`] = `
<div>
  <div
    class="flex items-center justify-center p-8"
  >
    <div
      class="flex flex-col items-center gap-1 text-center"
    >
      <h3
        class="text-3xl font-bold tracking-tight"
      >
        401 Unauthorized
      </h3>
      <p
        class="text-muted-foreground"
      >
        Please login to continue
      </p>
      <a
        href="/login"
      >
        <button
          class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-10 px-4 py-2 mt-4"
        >
          Login
        </button>
      </a>
    </div>
  </div>
</div>
`;
